### PR TITLE
chore: x/tokenfactory audit

### PR DESCRIPTION
**Pull request by alexanderbez**
Closes: #1914

## What is the purpose of the change

Auditing the `x/tokenfactory` module

## Brief Changelog

* Revising proto comments
* Revising keeper and types godocs
* Revising tests

### CODE REVIEW

**Comment by alexanderbez** [comment]
Not sure why, but after modifying the proto schemas and running `make proto-gen`, I get no updated compiled files...

**Comment by p0mvn** [comment]
> Not sure why, but after modifying the proto schemas and running `make proto-gen`, I get no updated compiled files...

I tried running `make proto-all` on my side and saw some updates to `pb.go` files.

Maybe try `make proto-all` instead of `proto-gen`? If it doesn't generate anything in your local environment, please let me know. I can try to make a commit from my side to see if that fixes it

File: proto/osmosis/tokenfactory/v1beta1/tx.proto [hunk 1]
```diff
@@ -61,7 +64,16 @@ message MsgBurn {
...
-// // ===================== MsgForceTransfer
+// MsgChangeAdmin is the sdk.Msg type for allowing an admin account to reassign
+// adminship of a denom to a new account
+message MsgChangeAdmin {
+  string sender = 1 [ (gogoproto.moretags) = "yaml:\"sender\"" ];
+  string denom = 2 [ (gogoproto.moretags) = "yaml:\"denom\"" ];
+  string new_admin = 3 [ (gogoproto.moretags) = "yaml:\"new_admin\"" ];
+}
+
+message MsgChangeAdminResponse {}
```
Comment (p0mvn) [inline proto/osmosis/tokenfactory/v1beta1/tx.proto:75]: nit: doc mesage for `MsgChangeAdminResponse`

File: x/tokenfactory/README.md [hunk 2]
```diff
@@ -11,33 +11,43 @@ created denom. Once a denom is created, the original creator is given
 - Mint their denom to any account
 - Burn their denom from any account
 - Create a transfer of their denom between any two accounts
-- Change the admin In the future, more admin capabilities may be
-    added. Admins can choose to share admin privileges with other
-    accounts using the authz module. The `ChangeAdmin` functionality,
-    allows changing the master admin account, or even setting it to
-    `""`, meaning no account has admin privileges of the asset.
-
+- Change the admin In the future, more admin capabilities may be added. Admins
```
Comment (p0mvn) [inline x/tokenfactory/README.md:14]:
```suggestion
- Change the admin. In the future, more admin capabilities may be added. Admins
```

File: x/tokenfactory/README.md [hunk 3]
```diff
@@ -68,15 +80,18 @@ message MsgBurn {
...
   - Check that the denom minting is created via `tokenfactory` module
   - Check that the sender of the message is the admin of the denom
 - Burn designated amount of tokens for the denom via `bank` module
 
-
 ### ChangeAdmin
-- Burning of a specific denom is only allowed for the creator of the denom registered during `CreateDenom`
-``` {.go}
+
+Burning of a specific denom is only allowed for the creator of the denom
```
Comment (p0mvn) [inline x/tokenfactory/README.md:91]: I think this text might be about `Burn`, not `ChangeAdmin`

File: x/tokenfactory/module.go [hunk 4]
```diff
@@ -149,7 +152,8 @@ func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, gs json.Ra
 	return []abci.ValidatorUpdate{}
 }
 
-// ExportGenesis returns the capability module's exported genesis state as raw JSON bytes.
+// ExportGenesis returns the capability module's exported genesis state as raw
```
Comment (p0mvn) [inline x/tokenfactory/module.go:155]:
```suggestion
// ExportGenesis returns the x/tokenfactory module's exported genesis state as raw
```

File: x/tokenfactory/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderbez) [reply x/tokenfactory/README.md]: AFAIK, current admin caps are just who created the token and that gives them the ability to perform certain actions. It does not extend beyond that AFAIK.

cc @sunnya97

**Comment by p0mvn** [comment]
> @p0mvn I tried both commands and I get no diff. I'm not sure but maybe it's because I'm on an m1 chip? Regardless, I have zero issues compiling proto in the SDK repo...

That's weird. Probably, a machine-specific issue. I made https://github.com/osmosis-labs/osmosis/pull/1971 to your PR to see if that fixes CI. This is what my Linux machine produces from your branch

**Comment by ValarDragon** [comment]
Did you find nothing odd in the logic / tests / events of token factory?

**Comment by alexanderbez** [comment]
> Did you find nothing odd in the logic / tests / events of token factory?

I think test coverage could improve, but what is there there now looks sufficient.

**Review by ValarDragon — APPROVED** [review]

File: x/tokenfactory/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunnya97) [reply x/tokenfactory/README.md]: The current admin capabilities are only minting and burning

**Pull request merged**